features/pipeline_parallel/resnet.py

```diff
@@ -165,10 +52,18 @@ def train():
     args = parser.parse_args()
     colossalai.launch_from_torch(backend=args.backend, config=CONFIG)
     logger = get_dist_logger()
+    pipelinable = PipelinableContext()
 
     # build model
-    model = resnet50()
-    model = build_pipeline_model(model, num_chunks=NUM_CHUNKS, verbose=True)
+    with pipelinable:
+        model = resnet50()
+
+    exec_seq = ['conv1', 'bn1', 'relu', 'maxpool', 'layer1.0', 'layer1.1', 'layer1.2', \
+                'layer2.0', 'layer2.1', 'layer2.2', 'layer2.3', 'layer3.0', 'layer3.1', \
+                'layer3.2', 'layer3.3', 'layer3.4', 'layer3.5', 'layer4.0', 'layer4.1', \
+                'layer4.2', 'avgpool', (lambda x: torch.flatten(x, 1), "behind"), 'fc']
```

Comment:
I would recommend not to go into the layer sub-index (i.e. use layer2 instead of layer 2.0 - 2.3). This is because that in the torchvision implementation, resnet forward is conducted in the unit of layer only.

<img width="357" alt="Screenshot 2022-05-06 at 3 30 07 PM" src="https://user-images.githubusercontent.com/31818963/167087277-45a722e0-5723-410a-a5f6-dd49e48e9b89.png">